src/main/cpp/profiler.h

```diff
@@ -79,12 +90,18 @@ class Profiler {
 
     Processor *processor;
 
-    SignalHandler handler_;
+    SignalHandler* handler_;
+
+    bool reloadConfig;
```

Comment:
I haven't realized that multiple Java threads can call `Agent` methods. Though Agent API is currently `synchronized` in Java, it doesn't guarantee anything about variables state. I think there're 2 possible solutions:
1. Assuming Java Agent API will always be synchronized and none of Profiler methods can be called simultaneously, we can add Acquire/Release memory fences in appropriate places - `std::atomic_thread_fence`.
2. We can synchronize `Profiler` instead by adding atomic flag that will be CASed for the duration of configuration actions. Flag should also have Acquire/Release semantic, so that all gets can do Acquire fence before returning any values. This can solve a lot of issues, like starting profiler from controller and Java code at the same time, or concurrent modifications.